backend/cn/wellTyped.ml

```diff
@@ -437,6 +437,14 @@ module WIT = struct
             let@ () = ensure_arith_type ~reason:loc t in
             let@ t' = check (IT.loc t) (IT.bt t) t' in
             begin match IT.bt t, is_const t' with
+            | Integer, Some (Z z', _) when Z.equal z' Z.zero ->
+               let msg =
+                  !^"Division" ^^^ squotes (IT.pp (div_ (t,t') loc))
+                  ^^^ !^"cannot have 0 as its right-hand argument."
+                  ^^^ !^"Treating as uninterpreted."
+                in
+                warn loc msg;
+                failwith "Division by Zero"
```

Comment:
This is already covered by the subsequent check `Z.leq z' Z.zero`. I think the comment means that you have to prove that the divisor is positive in `check.ml` by using the SMT solver (e.g. calling `provable` ).